e2e/stop_test.go

```diff
@@ -0,0 +1,344 @@
+
+		clusteringMf := mfs["moco_cluster_clustering_stopped"]
+		Expect(clusteringMf).NotTo(BeNil())
+		clusteringMetric := findMetric(clusteringMf, map[string]string{"namespace": "stop", "name": "test"})
+		Expect(clusteringMetric).NotTo(BeNil())
+		Expect(clusteringMetric.GetGauge().GetValue()).To(BeNumerically("==", 0))
+
+		reconcileMf := mfs["moco_cluster_reconciliation_stopped"]
+		Expect(reconcileMf).NotTo(BeNil())
+		reconcileMetric := findMetric(reconcileMf, map[string]string{"namespace": "stop", "name": "test"})
+		Expect(reconcileMetric).NotTo(BeNil())
+		Expect(reconcileMetric.GetGauge().GetValue()).To(BeNumerically("==", 0))
+	})
+
+	It("stopped metrics", func() {
+		kubectlSafe(nil, "moco", "stop", "clustering", "test", "-n", "stop")
+		kubectlSafe(nil, "moco", "stop", "reconciliation", "test", "-n", "stop")
+
+		Eventually(func() error {
+			cluster, err := getCluster("stop", "test")
+			if err != nil {
+				return err
+			}
+			reconcileCond := meta.FindStatusCondition(cluster.Status.Conditions, mocov1beta2.ConditionReconciliationActive)
+			if reconcileCond.Status == metav1.ConditionTrue {
+				return fmt.Errorf("reconciliation is active: %s", reconcileCond.Status)
+			}
+			clusteringCond := meta.FindStatusCondition(cluster.Status.Conditions, mocov1beta2.ConditionClusteringActive)
+			if clusteringCond.Status == metav1.ConditionTrue {
+				return fmt.Errorf("clustering is active: %s", clusteringCond.Status)
+			}
+			return nil
+		}).Should(Succeed())
+
+		out := kubectlSafe(nil, "-n", "moco-system", "get", "pods", "-l", "app.kubernetes.io/component=moco-controller", "-o", "json")
+		pods := &corev1.PodList{}
+		err := json.Unmarshal(out, pods)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(pods.Items).To(HaveLen(1))
+		addr := pods.Items[0].Status.PodIP
+		out, err = runInPod("curl", "-sf", fmt.Sprintf("http://%s:8080/metrics", addr))
+		Expect(err).NotTo(HaveOccurred())
+
+		mfs, err := (&expfmt.TextParser{}).TextToMetricFamilies(bytes.NewReader(out))
+		Expect(err).NotTo(HaveOccurred())
+
+		clusteringMf := mfs["moco_cluster_clustering_stopped"]
+		Expect(clusteringMf).NotTo(BeNil())
+		clusteringMetric := findMetric(clusteringMf, map[string]string{"namespace": "stop", "name": "test"})
+		Expect(clusteringMetric).NotTo(BeNil())
+		Expect(clusteringMetric.GetGauge().GetValue()).To(BeNumerically("==", 1))
+
+		reconcileMf := mfs["moco_cluster_reconciliation_stopped"]
+		Expect(reconcileMf).NotTo(BeNil())
+		reconcileMetric := findMetric(reconcileMf, map[string]string{"namespace": "stop", "name": "test"})
+		Expect(reconcileMetric).NotTo(BeNil())
+		Expect(reconcileMetric.GetGauge().GetValue()).To(BeNumerically("==", 1))
+
+		healthyMf := mfs["moco_cluster_healthy"]
+		Expect(healthyMf).NotTo(BeNil())
+		healthyMetric := findMetric(healthyMf, map[string]string{"namespace": "stop", "name": "test"})
+		Expect(healthyMetric).NotTo(BeNil())
+		Expect(math.IsNaN(healthyMetric.GetGauge().GetValue())).To(BeTrue())
```

Comment:
Is it too redundant to test that `moco_cluster_available`, `moco_cluster_ready_replicas`, `moco_cluster_errant_replicas` are also `NaN`?

Reply:
https://github.com/cybozu-go/moco/pull/578/commits/fb35e368e51a2ad9c8b4a48ab56747f2fa048777

I have added additional tests regarding metrics.